tests/integration/http.rs

```diff
@@ -28,36 +39,364 @@ async fn test_unknown_parent_hash() {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn test_wrong_hash() {
-    let client = get_client().await;
+async fn test_valid_block() {
+    let provider = MockEthProvider::default();
+    let client = get_client(Some(provider.clone())).await;
+    let validation_request_body: ValidationRequestBody = generate_valid_request(provider, None);
+
+    let result = ValidationApiClient::validate_builder_submission_v2(
+        &client,
+        validation_request_body.clone(),
+    )
+    .await;
+    assert!(result.is_ok());
+}
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_block_number_already_known() {
+    let provider = MockEthProvider::default();
+    let client = get_client(Some(provider.clone())).await;
     let validation_request_body: ValidationRequestBody =
-        serde_json::from_str(VALIDATION_REQUEST_BODY).unwrap();
-    let old_timestamp = format!("{:}", validation_request_body.execution_payload.timestamp);
-    let new_timestamp = "1234567";
+        generate_valid_request(provider.clone(), None);
+    let block = try_into_block(
+        validation_request_body.execution_payload.clone().into(),
+        None,
+    )
+    .expect("failed to create block");
+    provider.add_block(B256::random(), block);
+    // Double check that the header for this number is known
+    assert!(provider
+        .header_by_number(validation_request_body.execution_payload.block_number)
+        .unwrap()
+        .is_some());
+
+    let result = ValidationApiClient::validate_builder_submission_v2(
+        &client,
+        validation_request_body.clone(),
+    )
+    .await;
+    // TODO: Verify that this is expected behaviour (if not check if specific to mock provider)
```

Comment:
Is this expected behaviour ? I.e. the validation should be successful if a block of the same (or higher) block number (but with different hash) already exists on chain ?

Reply:
Makes sense to me if the block is otherwise valid. It's up to the builder / proposer to decide whether the chain should fork and a block with the same block number get published.

What does the flashbots code do?

Reply:
I didn't find any code checking wether the given block number is already known in the flahbots code. 

I did find code in the [VerifyHeader](https://github.com/flashbots/builder/blob/03ee71cf0a344397204f65ff6d3a917ee8e06724/consensus/beacon/consensus.go#L90) function that from my understanding should check wether the parent block's number is one less than the number in the current payload. 
I will double check what the behaviour is in our case.

Reply:
Added a test that verifies rejection if the parent block umber doesnt match:
https://github.com/ultrasoundmoney/reth-payload-validator/pull/11/commits/4f0eede0e584cfeb36e2e734226551ba49166818